pelican/generators.py

```diff
@@ -556,7 +553,9 @@ def generate_context(self):
             self.add_source_path(article)
 
         self.articles, self.translations = process_translations(
-            all_articles,
+            all_articles, 'category')
+        self.articles = order_content(
+            self.articles,
             order_by=self.settings['ARTICLE_ORDER_BY'])
         self.drafts, self.drafts_translations = \
             process_translations(all_drafts)
```

Comment:
You forgot to order `drafts` (or `hidden_pages` below).

Reply:
Yes they were. `process_translations` did that.

Reply:
`process_translations` only did that if the `order_by` parametre was true. For drafts and hidden pages, `order_by` defaulted to None.

Actually, I have to apologise for the presentation here. This is part of the first commit, which has already been merged as a separate PR (which at the time you ok-ed). I am sorry for letting you look at this twice.

Reply:
Well, scratch that. I'm not good at reading tonight. For some reason, I read as if this commit removed the first sort (by `slug`), which is not the case. So all good here :).

Reply:
Well, thinking about it again, even though it wasn't like that previously, wouldn't it make sense to *order* `drafts` and `hidden_pages` same as `articles` or `pages`?

Reply:
I was wondering about that as well (it took me ages to recall why I wrote it like this). I guess they aren't sorted now because they don't feature on the `index` page and on the various category, tag and author pages? But I guess nothing stops people from creating a page with a list of drafts, and then they would want those sorted.